api/pkg/cli/hub/get_resource.go

```diff
@@ -365,8 +413,50 @@ func (rr *TektonHubResourceResult) MinPipelinesVersion() (string, error) {
 	return *rr.resourceData.LatestVersion.MinPipelinesVersion, nil
 }
 
-// MinPipelinesVersion returns the minimum pipeline version the resource is compatible
+// MinPipelinesVersion returns the minimum pipeline version the resource is compatible from Artifact Hub
 func (rr *ArtifactHubResourceResult) MinPipelinesVersion() (string, error) {
-	//TODO
-	return "", nil
+	if err := rr.validateData(); err != nil {
+		return "", err
+	}
+	resp := ArtifactHubPkgResponse{}
+	if err := json.Unmarshal(rr.data, &resp); err != nil {
+		return "", err
+	}
+	if resp.Data.PipelineMinVer == "" {
+		return "", fmt.Errorf("min pipeline version is not specified in the resource")
+	}
+	return resp.Data.PipelineMinVer, nil
+}
+
+func (rr *ArtifactHubResourceResult) validateData() error {
+	if rr.err != nil {
+		return rr.err
+	}
+	if rr.set {
+		return nil
+	}
+
+	if rr.status == http.StatusNotFound {
+		return fmt.Errorf("No Resource Found")
+	}
+
+	return nil
+}
+
+func findArtifactHubVerions(data []byte) ([]string, error) {
+	resp := ArtifactHubPkgResponse{}
+	if err := json.Unmarshal(data, &resp); err != nil {
+		return nil, fmt.Errorf("error unmarshalling json response: %w", err)
+	}
+	if len(resp.AvailableVersions) == 0 {
+		return nil, fmt.Errorf("no available versions found in Artifact Hub")
+	}
+
```

Comment:
If there is a single version should we just return that version instead of looping it

Reply:
Hmm, I think the loop already covers 1-version scenario. Looks like redundant code to have an extra if statement... 🤔